lib/src/controllers/player_controller.dart

```diff
@@ -130,7 +130,10 @@ class PlayerController extends ChangeNotifier {
     bool shouldExtractWaveform = true,
     int noOfSamples = 100,
   }) async {
-    path = Uri.parse(path).path;
+    if (!path.startsWith('http')) {
+      // Keep the full URL for remote files and strip for local files
+      path = Uri.parse(path).path;
```

Comment:
Can you please use `tryParse` and if it returns null then throw exception so that it doesn't crash from native.